### PR TITLE
Update to the latest wdlTools (0.3.1)

### PR DESCRIPTION
**Pull request by jdidion**
* TextSource renamed to SourceLocation
* The 'text' fields on the AST have been updated to 'loc'
* Implement DxFileAccessProtocol
* Use FileResolver instead of SourceCode
* Get rid of DxIoFunctions - this is being replaced by FileResolver - the replacement will be completed in another PR

### CODE REVIEW

**Comment by jdidion** [comment]
This PR comes first

File: src/main/scala/dx/core/languages/wdl/ParseSource.scala [hunk 1]
```diff
@@ -53,15 +54,14 @@ case class ParseSource(logger: Logger) {
...
-        val absPath: Path = WdlUtil.getLocalPath(url)
+    val (sources, adjunctFiles) = doc.source match {
+      case localFs: LocalFileSource =>
+        val absPath: Path = localFs.localPath
         val sources = Map(absPath.toString -> doc)
         val adjunctFiles = Adjuncts.findAdjunctFiles(absPath)
         (sources, adjunctFiles)
-      case Some(url) =>
-        val sources = Map(url.toString -> doc)
+      case fs =>
```
Comment (commandlinegirl) [inline src/main/scala/dx/core/languages/wdl/ParseSource.scala:63]: What is `fs`?

File: src/test/scala/dx/exec/WfFragRunnerTest.scala [hunk 2]
```diff
@@ -26,25 +26,32 @@ class WfFragRunnerTest extends AnyFlatSpec with Matchers {
     DxInstanceType("mem_ssd_unicorn", 100, 100, 4, 1.00f, Vector(("Ubuntu", "16.04")), gpu = false)
   private val instanceTypeDB = InstanceTypeDB(pricingAvailable = true, Vector(unicornInstance))
 
-  private def setup(): (DxPathConfig, DxFileAccessProtocol) = {
+  private def setup(): (DxPathConfig, FileSourceResolver) = {
     // Create a clean directory in "/tmp" for the task to use
-    val jobHomeDir: Path = Paths.get("/tmp/dxwdl_applet_test")
-    SysUtils.deleteRecursive(jobHomeDir.toFile)
-    SysUtils.safeMkdir(jobHomeDir)
+    val jobHomeDir: Path = Files.createTempDirectory("dxwdl_applet_test")
```
Comment (commandlinegirl) [inline src/test/scala/dx/exec/WfFragRunnerTest.scala:31]: In the next PR, in [TaskRunnerTest](https://github.com/dnanexus/dxWDL/pull/430/files#diff-001802157ca7e83cd5725af52b1e28a6R138) the deleteRecursive() and createDirectories() methods are called. Why they are not called here?

**Comment by commandlinegirl** [comment]
@jdidion When you add more changes or address comments to the PR, please submit a request for re-review of the changes  (so that we don't have unreviewed code in our codebase). Thanks!

File: src/main/scala/dx/core/languages/wdl/ParseSource.scala (hunk 1 above)
Comment (jdidion) [reply src/main/scala/dx/core/languages/wdl/ParseSource.scala:63]: abbreviation for FileSource - I changed it to be more clear

File: src/test/scala/dx/exec/WfFragRunnerTest.scala (hunk 2 above)
Comment (jdidion) [reply src/test/scala/dx/exec/WfFragRunnerTest.scala:31]: createTempDir creates a new temp directory - there is no need to delete it and recreate it as there will never be a collision. I'll make sure it's fixed in TaskRunnerTest too.

**Pull request merged**